packages/packages/locations/src/injections.tsx

```diff
@@ -0,0 +1,69 @@
+import { Location, Filler, Injection, InjectionOptions, Name } from './types';
+import FillWrapper from './components/filler-wrapper';
+
+const DEFAULT_PRIORITY = 10;
+
+let injections: Map<string, Injection> = new Map();
+
+type InjectIntoArgs = {
+	location: Location;
+	filler: Filler;
+	name: Name;
+	options?: InjectionOptions;
+}
+
+export function injectInto( { location, filler, name, options = {} }: InjectIntoArgs ) {
+	const id = generateId( location, name );
+
+	const existingInjection = injections.get( id );
+
+	if ( existingInjection && ! options?.overwrite ) {
+		// eslint-disable-next-line no-console
+		console.error(
+			`An injection named "${ name }" under location "${ location }" already exists. Did you mean to use "options.overwrite"?`
+		);
+
+		return;
+	}
+
+	const fallbackPriority = existingInjection
+		? existingInjection.priority
+		: DEFAULT_PRIORITY;
```

Comment:
why do we need the "fallback priority'? it changes the meaning of the (absence of) `priority` option, which might be a bit unexpected to users

Reply:
Removed